### ISSUE TITLE
Macro results vary when calling it via pyimagej 

### ISSUE THREAD

**Issue by haesleinhuepf** [imagej/pyimagej#148]
Repeatedly calling the same macro code from pyimagej leads to different results.

See detailed description: https://forum.image.sc/t/macro-results-vary-when-calling-it-via-pyimagej/57744

**Comment by ctrueden** [comment]
The Troubleshooting doc now mentions this issue as of e1b255450942dac1ea9a3e3f3afa85ee225c89b0. But no one has had time to solve it, unfortunately.

**Comment by imagejan** [comment]
I commented on the forum thread with some insights:

> For what it’s worth, I was able to reproduce the issue within Fiji’s script editor (i.e., independent of [#pyimagej](https://forum.image.sc/tag/pyimagej)). Running the following macro code repeatedly:
> 
> ```javascript
> run("Blobs (25K)");
> setAutoThreshold("Otsu");
> setOption("BlackBackground", true);
> run("Convert to Mask");
> run("Analyze Particles...", "  show=[Count Masks]");
> getStatistics(area, mean, min, max, std, histogram);
> print(max);
> ```
> 
> … will print ever increasing values. It seems like there’s a counter for the Count Masks images that doesn’t get reset properly.
> 
> Judging from this line:
> https://github.com/imagej/ImageJ/blob/152c3d3def253a9c55f03214995b683f6f7008eb/ij/plugin/filter/ParticleAnalyzer.java#L1009-L1009
> 
> 
>     			case ROI_MASKS: drawRoiFilledParticle(drawIP, roi, mask, rt.size()); break;
> 
> … the value for each drawn ROI is determined from the size of the internal `ResultsTable` (`rt.size()`).
> 
> ---
> 
> I can work around the issue by adding `Table.reset("Results");` to my macro code:
> 
> ```javascript
> run("Blobs (25K)");
> setAutoThreshold("Otsu");
> setOption("BlackBackground", true);
> run("Convert to Mask");
> run("Analyze Particles...", "  show=[Count Masks]");
> getStatistics(area, mean, min, max, std, histogram);
> Table.reset("Results");
> print(max); // will now always print 64
> ```
>

**Comment by ctrueden** [comment]
@imagejan Would you consider this a bug in ImageJ? I.e. should we move this issue to imagej/ImageJ?

**Comment by imagejan** [comment]
> Would you consider this a bug in ImageJ?

Yes. I just tested it with plain ImageJ, and observe the same behavior as seen with Fiji.
